akka-sample-kafka-to-sharding-scala/processor/src/main/scala/sample/sharding/kafka/UserEventsKafkaProcessor.scala

```diff
@@ -1,63 +1,45 @@
 package sample.sharding.kafka
 
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.util.Try
+
 import akka.Done
+import akka.actor.Scheduler
 import akka.actor.typed.scaladsl.AskPattern._
 import akka.actor.typed.scaladsl.Behaviors
 import akka.actor.typed.scaladsl.adapter._
-import akka.actor.typed.{ActorRef, Behavior}
-import akka.actor.{ActorSystem, Scheduler}
+import akka.actor.typed.ActorRef
+import akka.actor.typed.Behavior
+import akka.actor.typed.{ActorSystem => TypedActorSystem}
 import akka.kafka.cluster.sharding.KafkaClusterSharding
-import akka.kafka.scaladsl.{Committer, Consumer}
-import akka.kafka.{CommitterSettings, Subscriptions}
+import akka.kafka.scaladsl.Committer
+import akka.kafka.scaladsl.Consumer
+import akka.kafka.CommitterSettings
+import akka.kafka.Subscriptions
 import akka.pattern.retry
+import org.slf4j.LoggerFactory
 import sample.sharding.kafka.serialization.UserPurchaseProto
 
-import scala.concurrent.duration._
-import scala.concurrent.{ExecutionContextExecutor, Future}
-import scala.util.Try
-
 object UserEventsKafkaProcessor {
 
   sealed trait Command
 
   private case class KafkaConsumerStopped(reason: Try[Any]) extends Command
 
+  private val logger = LoggerFactory.getLogger(this.getClass)
```

Comment:
Dedicated logger for the stream consumption. Stream log operator was not useful because doesn't allow logging custom message. 

I was looking for something like `withLogger` that would transform the element from `E` to `(E, Logger)` and then unzip back to `E`. Maybe something that we could addd to Akka Stream?